docs/build/abci/01-prepare-proposal.md

```diff
@@ -0,0 +1,43 @@
+## Prepare Proposal
+
+`PrepareProposal` handles construction of the block, meaning that when a proposer
+is preparing to propose a block, it requests the application to evaluate a
+`RequestPrepareProposal`, which contains a series of transactions from CometBFT's
+mempool. At this point, the application has complete control over the proposal.
+It can modify, delete, and inject transactions from it's own app-side mempool into
+the proposal or even ignore all the transactions altogether. What the application
+does with the transactions provided to it by `RequestPrepareProposal` have no
+effect on CometBFT's mempool.
+
+Note, that the application defines the semantics of the `PrepareProposal` and it
+MAY be non-deterministic and is only executed by the current block proposer.
+
+Now, reading mempool twice in the previous sentence is confusing, lets break it down.
+CometBFT has a mempool that handles gossiping transactions to other nodes
+in the network. How these transactions are ordered is determined by CometBFT's
+mempool, typically FIFO. However, since the application is able to fully inspect
```

Comment:
I think FIFO is the ONLY order (priority mempool in comet was removed/deprecated IIRC).